packages/commerce-server/src/determine-purchase-type.ts

```diff
@@ -41,81 +133,79 @@ type DeterminePurchaseTypeOptions = {
-
-  const user = !!email && (await getUserByEmail(email))
-  const {id: userId} = user || {}
-
-  // Find the purchase record associated with this Stripe Checkout Session
-  // (via the `stripeChargeId`).
-  const checkoutSessionPurchase = await getPurchaseForStripeCharge(
-    stripeChargeId,
-  )
-
-  // return early if we don't have a userId or a purchase corresponding to
-  // this checkout session
-  if (!userId || !checkoutSessionPurchase?.id) return null
-
-  const allUserPurchases = await getPurchasesForUser(userId)
-
-  const bulkPurchases = allUserPurchases.filter(
-    (purchase) => purchase.bulkCoupon !== null,
-  )
-  const individualPurchase = allUserPurchases.find(
-    (purchase) =>
-      purchase.bulkCoupon === null && purchase.redeemedBulkCouponId === null,
-  )
-
-  // if the user has no bulk purchases and their only individual purchase
-  // is the purchase for this checkout session, then this session's purchase
-  // type is NEW_INDIVIDUAL_PURCHASE.
-  if (
-    isEmpty(bulkPurchases) &&
-    match(individualPurchase?.id, checkoutSessionPurchase.id)
-  ) {
+  try {
+    // Grab the Stripe Charge ID associated with the completed checkout session
+    // so that we can reference the associated purchase in our database.
+    const purchaseInfo = await stripeData({
+      checkoutSessionId: checkoutSessionId as string,
+      stripeCtx,
+    })
+    const {email, stripeChargeId} = purchaseInfo
+
+    const user = !!email && (await getUserByEmail(email))
+    const {id: userId} = user || {}
+
+    // Find the purchase record associated with this Stripe Checkout Session
+    // (via the `stripeChargeId`).
+    const checkoutSessionPurchase = await getPurchaseForStripeCharge(
+      stripeChargeId,
+    )
+
+    // return early if we don't have a userId or a purchase corresponding to
+    // this checkout session
+    if (!userId || !checkoutSessionPurchase?.id)
+      throw new Error('Missing userId or purchase for checkout session')
+
+    const allUserPurchases = await getPurchasesForUser(userId)
+
+    const {type, data} = summarizePurchases(
+      checkoutSessionPurchase,
+      allUserPurchases,
+    )
+
+    if (type === 'BULK') {
+      const {
+        otherTeamBulkPurchases,
+        otherIndividualPurchases,
+        focalBulkPurchaseIsChronologicallyFirst,
+      } = data.focalProduct
+
+      if (
+        isEmpty(otherTeamBulkPurchases) ||
+        focalBulkPurchaseIsChronologicallyFirst
+      ) {
+        if (isEmpty(otherIndividualPurchases)) {
+          // this is the first purchase, it is a new bulk coupon purchase
+          return NEW_BULK_COUPON
+        } else {
+          // they made an individual purchase before, but are now purchasing
+          // a bulk coupon, that's an upgrade to bulk
+          return INDIVIDUAL_TO_BULK_UPGRADE
+        }
+      } else {
+        // anything else at this point is a purchase of additional bulk seats
+        return EXISTING_BULK_COUPON
```

Comment:
Able to handle the 3 bulk scenarios based on the summarized purchase data from `summarizePurchases`.